app/javascript/src/components/Dashboard/Contacts/Pane/Form.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+
+import { Formik, Form } from "formik";
+import { Button, Pane } from "neetoui/v2";
+import { Input, Select } from "neetoui/v2/formik";
+
+import formValidationSchemas from "constants/formValidationSchemas";
+
+import { ROLE_OPTS, FORM_INIT } from "./constants";
+
+export default function ContactForm({ onClose, setContacts }) {
+  const [submitted, setSubmitted] = useState(false);
+
+  const getFormattedDate = () => {
+    const options = { weekday: "short", day: "2-digit", year: "numeric" };
+    const stringDateArray = new Date()
+      .toLocaleDateString("en-US", options)
+      .split(" ");
+    return (
+      stringDateArray[1] + " " + stringDateArray[0] + ", " + stringDateArray[2]
+    );
+  };
+
+  const handleSubmit = values => {
+    setContacts(contacts => [
+      { ...values, role: values.role.label, createdAt: getFormattedDate() },
+      ...contacts,
+    ]);
+    setSubmitted(true);
+    onClose();
+  };
+
+  return (
+    <Formik
+      initialValues={FORM_INIT}
+      onSubmit={handleSubmit}
+      validateOnBlur={submitted}
+      validateOnChange={submitted}
+      validationSchema={formValidationSchemas.contactsForm}
```

Comment:
Please move validation schema and initial values to the scope of this directory.